PhotoFeed/app/src/main/java/com/example/yhuang/photofeed/MainActivity.java

```diff
@@ -0,0 +1,212 @@
+        // Handle presses on the action bar items
+        switch (item.getItemId()) {
+            case R.id.action_feed:
+                openMyFeed();
+                return true;
+            case R.id.action_search:
+                openSearch();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    /**
+     * the openSearch() method clears the current displaying data and opens the search page
+     */
+    public void openSearch() {
+        delete.setVisibility(View.GONE);
+        feed.setVisibility(View.VISIBLE);
+        editText.setVisibility(View.VISIBLE);
+        search.setVisibility(View.VISIBLE);
+        cur_page = 0;
+        items.clear();
+        imageview.setImageResource(0);
+    }
+
+    /**
+     * the openMyFeed() method clears the current displaying data and opens the feed page
+     */
+    public void openMyFeed() {
+        delete.setVisibility(View.VISIBLE);
+        feed.setVisibility(View.GONE);
+        editText.setVisibility(View.GONE);
+        search.setVisibility(View.GONE);
+        cur_page = 0;
+        items.clear();
+        imageview.setImageResource(0);
+
+        PhotoHelper.SyncWithDatabase(mDbHelper, items);
+        if (items.isEmpty()) {
+            Toast.makeText(getApplicationContext(), "You haven't feed anything yet", Toast.LENGTH_SHORT).show();
+        } else {
+            new ImageLoadTask(items.get(cur_page), imageview).execute();
+        }
+    }
+
+    /**
+     * the makeRequestWithCallback() takes a string and send a http request to google; assign the return links to an arraylist
+     */
+    public void makeRequestWithCallback(String searchText,final ArrayList<String> itemlist) {
+        HttpHandler handler = new HttpHandler(MainActivity.this);
+        handler.searchWithCallback(searchText, new SuccessCallback() {
+            @Override
+            public void callback(boolean success, JSONObject response) {
+                if (success) {
+                    Log.d("Success", Boolean.toString(success));
+                    try {
+                        JSONArray items = response.getJSONArray("items");
+
+                        itemlist.clear();
+                        cur_page = 0;
+
+                        for (int i = 0; i < items.length(); i++) {
+                            JSONObject itemObject = items.getJSONObject(i);
+                            String link = itemObject.getString("link");
+                            itemlist.add(link);
+                        }
+
+                        new ImageLoadTask(itemlist.get(0), imageview).execute();
+
+                    } catch (Exception err) {
+                        Log.e("Error: ", err.getMessage());
+                    }
+
+
+                } else {
+                    Log.d("Failure", Boolean.toString(success));
+                }
+            }
+        });
```

Comment:
Moving this callback to a seperate method such as parseJson would improve clarity.